job/service.go

```diff
@@ -738,3 +692,194 @@ func populateDownstreamDAGs(dagTree *tree.MultiRootTree, jobSpec models.JobSpec,
 
 	return rootNode, nil
 }
+
+// Refresh fetches all the requested jobs, resolves its dependencies, assign proper priority weights,
+// compile all jobs in the project and upload them to the destination store.
+func (srv *Service) Refresh(ctx context.Context, projectSpec models.ProjectSpec, namespaceJobNamePairs []models.NamespaceJobNamePair,
```

Comment:
I was limiting the usage of protobuf model to only in the handler and not passing it to the service layer. this `namespaceJobNamePairs ` is only being used to get all of the jobSpecs for dependency resolution. should we do it early in the handler? what do you think?

Reply:
As discussed, still use `NamespaceJobNamePair` to avoid using external contract in the service layer. However, we are not passing NamespaceSpec, but NamespaceName instead (and fetching the spec just before where it is being used).